inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_quantization.cpp

```diff
@@ -20,415 +20,412 @@
+void  prepare_quantization::prepare_scale_shift_opt(program_impl &p, quantize_node& quantize_node) {
+    const auto& stream = p.get_stream();
+    program_node &input_low_node = quantize_node.get_dependency(1);
+    program_node &input_high_node = quantize_node.get_dependency(2);
+    program_node &output_low_node = quantize_node.get_dependency(3);
+    program_node &output_high_node = quantize_node.get_dependency(4);
 
-            program_node &input_low_node = quantize_node.get_dependency(1);
-            program_node &input_high_node = quantize_node.get_dependency(2);
+    if (!input_low_node.is_type<data>() || !input_high_node.is_type<data>() ||
+        !output_low_node.is_type<data>() || !output_high_node.is_type<data>()) {
+        return;
+    }
 
-            if (!input_low_node.is_type<data>() || !input_high_node.is_type<data>()) {
-                return;
+    auto &input_low = input_low_node.as<data>();
+    auto &input_high = input_high_node.as<data>();
+    auto &output_low = output_low_node.as<data>();
+    auto &output_high = output_high_node.as<data>();
+
+    auto mem_input_low = input_low.get_attached_memory_ptr();
+    auto mem_input_high = input_high.get_attached_memory_ptr();
+    auto mem_output_low = output_low.get_attached_memory_ptr();
+    auto mem_output_high = output_high.get_attached_memory_ptr();
+
+    auto scales_layout = mem_input_low->get_layout();
+    scales_layout.size = tensor::max(scales_layout.size, mem_input_high->get_layout().size);
+    scales_layout.size = tensor::max(scales_layout.size, mem_output_low->get_layout().size);
+    scales_layout.size = tensor::max(scales_layout.size, mem_output_high->get_layout().size);
+
+    auto mem_input_scale  = p.get_engine().allocate_memory(scales_layout, false);
+    auto mem_input_shift  = p.get_engine().allocate_memory(scales_layout, false);
+    auto mem_output_scale = p.get_engine().allocate_memory(scales_layout, false);
+    auto mem_output_shift = p.get_engine().allocate_memory(scales_layout, false);
+
+    auto get_offset_safe = [](const layout& l, const tensor& idx) -> int {
+        auto sizes = l.size;
+        auto pitches = l.get_pitches();
+
+        return (idx.batch[0] % sizes.batch[0])*pitches.batch[0]
+                        + (idx.feature[0] % sizes.feature[0])*pitches.feature[0]
+                        + (idx.spatial[1] % sizes.spatial[1])*pitches.spatial[1]
+                        + (idx.spatial[0] % sizes.spatial[0])*pitches.spatial[0];
+    };
+
+    auto lock_memory = [&stream] (memory::ptr memory, std::function<void(std::size_t, float)>& set_data,
+                           std::function<float(size_t)>& get_data) {
+        switch (memory->get_layout().data_type) {
+            case data_types::f32: {
+                mem_lock<float> data_lock{memory, stream};
+                float* data = data_lock.data();
+                set_data = [data] (size_t idx, float value) {
+                    data[idx] = value;
```

Comment:
This pointer is invalidated once `data_lock` is removed. So I believe this part must be reimplemented.

Reply:
Fixed